src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/validation/JsonIncludeObjectHavingNonNullFieldOrNonEmptyList.java

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2023 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.messaging.validation;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.List;
+
+/**
+ * This class is a custom Jackson's JsonInclude validation intended to exclude from JSON
+ * an object containing only null fields and non-null but empty Lists.
+ */
+public final class JsonIncludeObjectHavingNonNullFieldOrNonEmptyList {
```

Comment:
Looking at the class usage, I'd rename it to something closer to class description, i.e. `ExcludeObjectsWithOnlyNullFieldsOrEmptyLists`.
I don't think we should mention `JacksonInclude` mechanism in class name if we don't implement this interface.

Reply:
In fact, they made it in Jackson in a way that you don't need any interface (in my opinion, that's weird, but it is like it is). In Jackson, they have a hidden requirement to the equals method and naming/idea of include with negation (i.e. include NON_NULL), so I just followed something people could find on web. 
Frankly, I don't think some names would help here. I'm sure people would be confused anyway whether equals returns true means include or exclude. I can create an interface for it with a single usage for it or rename it,  but then on the usage side it's going to be 
```java
@JsonInclude(value = JsonInclude.Include.CUSTOM, valueFilter = ExcludeObjectsWithOnlyNullFieldsOrEmptyLists.class)
```
kind of still confusing.
So, what is it you proposing here? Maybe I can add some more text explanation

Reply:
I added a javadoc with the explanation.

Reply:
Fair enough.